code401Challenges/src/main/java/code401Challenges/tree/Tree.java

```diff
@@ -100,4 +99,26 @@ public Tree(Node<T> root) {
         return valuesArr;
     }
 
+    // ======= Breadth First Traversal, using a queue, for numbers
+    public void breadthFirstTraverse (Node<Integer> root) {
+        Queue<Integer> storageQ = new Queue<>();
+
+        if (root == null) {
+            throw new NoSuchElementException();
+        } else {
+            storageQ.enqueue(root.value);
```

Comment:
We don't just want to enqueue the value, but the whole Node, because when we take it out, we will need to enqueue it's left and right children

Reply:
Oh! Good point. Thank you!